### DISCUSSION TITLE
Obfuscating Source Code for production deployment

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#2208] [Q&A] [answered]
Hi, I currently have to deploy my Wave app in production in a company client's server due to data privacy concerns. I am looking for a wave to obfuscate or release Wave app script and run it whilst protecting proprietary code. I came across [this discussion](https://github.com/h2oai/wave/discussions/1363) as well. I am trying to understand this. How do I run obfuscated Wave app scripts?

**Comment by mturoci** [comment, marked as answer]
See [cythonization](https://cython.readthedocs.io/en/latest/src/quickstart/overview.html).

**Reply by n-srinidhi** [reply]
Thank you! How will I run an obfuscated Wave script?
can I run it as a normal wave script command? Like `wave run --no-reload app.py`

**Reply by mturoci** [reply]
Wave is just a python at the end of the day. The docs I linked above should have all the info you need.

**Reply by n-srinidhi** [reply]
Alright. Thanks a lot!
